lessons/git/git-collaboration-2in1/index.md

```diff
@@ -0,0 +1,291 @@
+
+Kdybyste měli v různých kopiích repozitáře zmatek, přijde vhod malé vysvětlení: jedna kopie je původní projekt na GitHubu, 
+kam správce projektu dává aktuální "oficiální verzi". 
+Další kopie na GitHubu je "vaše" a můžete si do ní nahrát cokoliv 
+(nejčastěji se v ní ale zveřejňují změny, které můžou být užitečné pro ostatní). 
+A třetí kopii repozitáře máte u sebe na počítači. 
+
+![Git workflow]({{ static('gh-workflow-diagram.svg') }})
+
+### Práce s lokálním repozitářem
+
+Než budeme schopní poslat nějaké změny k začlenění do našeho vybraného projektu, musíme se naučit, jak se s Gitem pracuje.
+Co je to vlastně ten zmíněný *repozitář*? Můžeme ho vytvořit z jakéhokoliv adresáře na disku, kde máme uložené soubory, 
+jen Gitu řekneme, že z něj chceme ten repozitář udělat. To má za následek:
+
+1. Git bude sledovat změny, které ve složce uděláme.
+2. Budeme schopni vytvořit verzi tohoto adresáře, ke které se můžeme kdykoliv vrátit.
+3. Můžeme nastavit vzdálené repozitáře, kam chceme změny nahrávat nebo naopak stahovat změny od jiných lidí. 
+
+!!! note "Pozor"
+    Budeme hodně pracovat s příkazovou řádkou. Jestli se s ní ještě nekamarádíš, koukni se na [úvod](../../beginners/cmdline/).
+    Nezapomeň: $ na začátku se nepíše; je tu proto, aby šlo poznat že jde o příkaz.
+
+Naším dnešním cílem je ale přispět do projektu, který založil někdo jiný. Proto použijeme příkaz `git clone` ke stažení vzdáleného repozitáře.
+
+    $ git clone https://github.com/asgeirrr/prezencka
+
+V aktuálním adresáři se nám vytvořil podadresář `prezencka`, který obsahuje všechny soubory, které vidíme ve webovém rozhraní na [GitHubu](https://github.com/asgeirrr/prezencka).
+
+
+#### Stav repozitáře (`git status`)
+
+Velmi často potřebujeme zjistit současný stav repozitáře. Navigujte v příkazové řádce do složky `prezencka` a zadejte `git status`. Git nám odpoví:
+
+    $ git status
+    On branch master
+    Your branch is up-to-date with 'origin/master'.
+    nothing to commit, working tree clean
+
+Git nám v tuto chvíli poskytl 3 informace:
+
+1. Že jsme na větvi `master`. Git umožňuje pracovat na více věcích současně pomocí tzn. *větví* mezi kterými můžeme přepínat. Pak se na na disku "objeví" verze souborů v dané větvi. My si dnes vystačíme s jednou hlavní větví, která se tradičně jmenuje `master`.
+2. Že naše větev `master` je aktuální vůči větvi na vzdáleném repozitáři (origin) na GitHubu.
+3. Že jsme neudělali žádné změny oproti poslední verzi souborů, což je pravda.
+
+#### První revize (`git commit`)
+
+Už jsme několikrát zmínili, že Git je verzovací systém. Jak že to ale soubory verzuje? 
+V podstatě mu musíme dát příkaz, aby zmrazil současný stav repozitáře, tzv. vytvořit *revizi*.
+Pak můžeme dělat další změny, ale už vždy budeme schopni se k této revizi vrátit.
+
+Přidejte do repozitáře soubor `vase_jmeno.txt`, např. tedy `magdalena_rettigova.txt`
```

Comment:
a) určitě, souhlasím
b) to bych nechal na nich, ať si to třeba udělají v Atomu, bohužel ten git bash se nezačal tenhle běh používat.